src/base/serial/tty_io.c

```diff
@@ -299,34 +299,39 @@ static int tty_already_locked(char *nam)
  */
 static int tty_lock(const char *path, int mode)
 {
-  char saved_path[strlen(config.tty_lockdir) + 1 +
-                  strlen(config.tty_lockfile) +
-                  strlen(path) + 1];
+  char *saved_path;
```

Comment:
Oh, that one is really nasty for
so many free() calls...
Just use PATH_MAX-sized buffer instead
of asprintf()?

Reply:
okay, done